ports/esp32/machine_timer.c

```diff
@@ -130,7 +112,33 @@ static mp_obj_t machine_timer_make_new(const mp_obj_type_t *type, size_t n_args,
     return self;
 }
 
-static void machine_timer_disable(machine_timer_obj_t *self) {
+machine_timer_obj_t *machine_timer_create(mp_uint_t timer) {
```

Comment:
I think this new create function should be used by the above `machine_timer_make_new`, because otherwise there's a lot of duplicated code.

Reply:
Done.

Reply:
For ESP32, SAMD, Renesas-RA and NRf there are still two commits. I can join them, leaving just one commit per port.

Reply:
Thanks for cleaning up the esp32 timer functions.  The diff is smaller now, which is great!